src/zenohRpcClient.cpp

```diff
@@ -126,90 +125,92 @@ std::future<UPayload> ZenohRpcClient::invokeMethod(const UUri &uri,
         spdlog::error("ZenohRpcClient is not initialized");
         return std::move(future);
     }
-    
+
     if (UMessageType::REQUEST != attributes.type()) {
         spdlog::error("Wrong message type = {}", UMessageTypeToString(attributes.type()).value());
         return std::move(future);
     }
-    
+
     auto uriHash = std::hash<std::string>{}(LongUriSerializer::serialize(uri));
+    auto header = MessageBuilder::buildHeader(attributes);
 
-    if (UMessageType::REQUEST != attributes.type()) {
-        spdlog::error("Wrong message type = {}", UMessageTypeToString(attributes.type()).value());
-        return std::move(future);
-    }
-  
-    auto message = MessageBuilder::build(attributes, payload);
-    if (0 == message.size()) {
-        spdlog::error("MessageBuilder failure");
-        return std::move(future);
-    }
+    z_owned_bytes_map_t map = z_bytes_map_new();
 
-    z_owned_reply_channel_t *channel = new z_owned_reply_channel_t;
+    z_bytes_t headerBytes = {.len = header.size(), .start = header.data()};
+    z_bytes_t payloadBytes = {.len = payload.size(), .start = payload.data()};
+    z_bytes_map_insert_by_alias(&map, z_bytes_new("header"), headerBytes);
+    z_bytes_map_insert_by_alias(&map, z_bytes_new("payload"), payloadBytes);
```

Comment:
you are not sending the payload as part of the header, it should be sent separately , same as you did for pub-sub